hi-ml-histopathology/testhisto/testhisto/datamodules/test_slides_datamodule.py

```diff
@@ -0,0 +1,107 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+import py
+import pytest
+import numpy as np
+
+from health_ml.utils.common_utils import is_gpu_available
+from testhisto.utils.utils_slides_datamodule import MockSlidesDataModule, MockWSIGenerator, MockWSIType
+
+
+no_gpu = not is_gpu_available()
+N_REPEAT_DIAG = 4
+
+
+@pytest.fixture(scope="session")
+def mock_wsi_root_dir(tmpdir_factory: pytest.TempdirFactory) -> py.path.local:
+    tmp_root_dir = tmpdir_factory.mktemp("mock_wsi")
+    wsi_generator = MockWSIGenerator(
+        tmp_path=tmp_root_dir,
+        mock_type=MockWSIType.PATHMNIST,
+        batch_size=1,
+        n_samples=4,
+        n_repeat_diag=N_REPEAT_DIAG,
+        n_repeat_tile=2,
+        n_channels=3,
+        n_levels=3,
+        tile_size=28,
+        background_val=255
+    )
+    wsi_generator.generate_mock_wsi()
+    return tmp_root_dir
+
+
+@pytest.mark.skipif(no_gpu, reason="Test requires GPU")
+@pytest.mark.gpu
+def test_tiling_on_the_fly(mock_wsi_root_dir: py.path.local) -> None:
+    batch_size, tile_count, tile_size, level, channels = 1, 16, 28, 0, 3
+    datamodule = MockSlidesDataModule(
+        root_path=mock_wsi_root_dir, batch_size=batch_size, tile_count=tile_count, tile_size=tile_size, level=level
+    )
+    dataloader = datamodule.train_dataloader()
+    for sample in dataloader:
+        # sanity check for expected shape
+        tiles, wsi_id = sample["image"], sample["slide_id"][0]
```

Comment:
I always suggest avoiding hard-coded string - they make maintenance a nightmare,  you can use `SildesDataset.IMAGE_COLUMN` and ``SildesDataset.SLIDE_ID_COLUMN``

Reply:
It applies also the hard-coded strings in the rest of this module